packages/support/lib/logging.js

```diff
@@ -116,10 +85,41 @@ function getLogger(prefix = null) {
   return /** @type {AppiumLogger} */ (wrappedLogger);
 }
 
-// export a default logger with no prefix
-const log = getLogger();
+/**
+ *
+ * @returns {[import('@appium/logger').Logger, boolean]}
+ */
+function _getLogger() {
+  // check if the user set the `_TESTING` or `_FORCE_LOGS` flag
+  const testingMode = process.env._TESTING === '1';
+  const forceLogMode = process.env._FORCE_LOGS === '1';
+  // if is possible that there is a logger instance that is already around,
+  // in which case we want t o use that
+  const useGlobalLog = !!global._global_npmlog;
+  const logger = testingMode && !forceLogMode
+    // in testing mode, use a mock logger object that we can query
+    ? MOCK_LOG
+    // otherwise, either use the global, or a new `npmlog` object
+    : (global._global_npmlog || globalLog);
+  // The default value is 10000, which causes excessive memory usage
+  logger.maxRecordSize = MAX_LOG_RECORDS_COUNT;
```

Comment:
I guess this is now also being set if `testingMode && !forceLogMode == true`, but I'm not sure if this will affect anything

Reply:
it does not make any difference as MOCK_LOG is just a noop wrapper